### ISSUE TITLE
Conditional simulator

### ISSUE THREAD

**Issue by NewbieXII** [kazuhikoarase/simcirjs#23]
can u create a simulator with condition where only specific device can be found in the workspace?
help me out :( 
i have been doing code and i can't perfect it. :(

**Comment by kazuhikoarase** [comment]
Hi,
It mean that you don't want full devices?

**Comment by NewbieXII** [comment]
your system is ok... it was awesome.. but it occur to my mind what if i want a quiz based instead freestyle simulating.
i just want to add a condition based where if the connection is correct or not correct based on given question... you know like teaching based.

**Comment by NewbieXII** [comment]
sorry ..just newbie in this programming but i want to become programmer just like you

**Comment by kazuhikoarase** [comment]
Could you send me a drawing your imaging?

**Comment by kazuhikoarase** [comment]
It can paste here like this.
![image](https://user-images.githubusercontent.com/1137546/189923354-fc1fbbf2-d7af-4647-a32a-b1d985274fff.png)

**Comment by NewbieXII** [comment]
hmm...... like this
GIven Instruction is here

device here and workspace is here

on the bottom of these there is button where will dictate if your work in your working space is correct based on the given instruction. 
if the instruction is not follow then the it will trigger that there is something wrong to your work
like you are answering a question to a quiz. if your answer is correct then you can go to the next number if your answer is wrong then you need to check your work... like that ..  i hope you understand.. sorry for my english

**Comment by NewbieXII** [comment]
![Untitled](https://user-images.githubusercontent.com/54292058/189935535-70aafccb-e3de-4430-b25f-7983b4ae86a3.png)


example in given instuction... you need to put LED,DC, and connect them. if the user click the button and follow the given instruction and pop you are correct then the user do another one. just like that hehe

**Comment by NewbieXII** [comment]
sorry for my english i hope i deliver my message correctly.. i really sorry.

**Comment by kazuhikoarase** [comment]
What's this button's label? I could not read it.
SVD ?
![image](https://user-images.githubusercontent.com/1137546/190131279-4bc79f35-be69-48b8-ab9f-f5fa38983e55.png)

**Comment by NewbieXII** [comment]
that button is submit

**Comment by kazuhikoarase** [comment]
Is your image as follows?

Step1. First exam.
![image](https://user-images.githubusercontent.com/1137546/190188926-5d2cd9a9-47d0-4766-b97e-c2d0b26ea59f.png)
Step2. User solve it.
![image](https://user-images.githubusercontent.com/1137546/190189536-6230d316-482c-4ab3-aaa4-71b96acab745.png)
Step3. Confirm it.
![image](https://user-images.githubusercontent.com/1137546/190192011-adee0570-c617-4ba6-9703-8f40e21363dc.png)
Step4. Go to next exam.
![image](https://user-images.githubusercontent.com/1137546/190195270-70b89698-e4ed-4498-9264-30a95d81779e.png)

**Comment by NewbieXII** [comment]
yeah like that... can i have the code? PLEASEEE :(((

**Comment by NewbieXII** [comment]
i have been struggle in doing that ... :( help mee please.

**Comment by kazuhikoarase** [comment]
Current source is here.
But it's hard to implement to check why a current circuit is correct.
```html
<!doctype html>
<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html;charset=UTF-8" />
    <script type="text/javascript" src="simcir.js"></script>
    <link rel="stylesheet" type="text/css" href="simcir.css" />
    <script type="text/javascript" src="simcir-basicset.js"></script>
    <link rel="stylesheet" type="text/css" href="simcir-basicset.css" />
    <script type="text/javascript" src="simcir-library.js"></script>
    <style>
.frame {
  padding: 16px;
  display: inline-block;
  border: 1px solid #000;
}
.instuction {
  padding: 4px 8px;
  margin: 8px 4px;
  display: inline-block;
  background-color: #ccc;
  font-size: 16pt;
}
.button {
  font-size: 16pt;
}
    </style>
    <title></title>
  </head>
  <body>
  <div class="frame">
    <div class="instuction">Instuction here</div>
    <div class="simcir">
    <!-- paste here your circuit data -->
{
  "width":400,
  "height":200,
  "showToolbox":true,
  "toolbox":[
    {"type":"DC"},
    {"type":"LED"}
  ],
  "devices":[
  ],
  "connectors":[
  ]
}
    </div>
    <button class="button" onclick="alert('Correct!')"> SUBMIT </button>
  </div>
  </body>
</html>
```

**Comment by JamesNewton** [comment]
Is it enough to check that it gives the correct output for all possible inputs?

**Comment by JamesNewton** [comment]
I am interested in your simulator having this ability as it would be useful for teaching. Like NandGame. But with your nicer user interface.

**Comment by kazuhikoarase** [comment]
@JamesNewton 
Where is lacking of user interface?

**Comment by NewbieXII** [comment]
yeah.. so hard :(... that is why i ask you ? hehehe cause i doing it for a month to do that... i have a idea,,, how about we use can the  connector as data finder?... something like this... we just want to know that the device is in the workspace.

"devices":[
  ],
  "connectors":[
  ]
  can we use these two as data finder?...

**Comment by JamesNewton** [comment]
@kazuhikoarase I'm saying your user interface is better.

**Comment by NewbieXII** [comment]
i need a condition to know that can tell what inside of the workspace. you know like evaluating what inside of the breadboard.
:(
